occurrence-clustering/src/main/java/org/gbif/occurrence/clustering/OccurrenceRelationships.java

```diff
@@ -176,27 +178,22 @@ private static void compareIdentifiers(OccurrenceFeatures o1, OccurrenceFeatures
     }
   }
 
-  static boolean equalsAndNotNull(OccurrenceFeatures o1, OccurrenceFeatures o2, String field) {
-    return o1.get(field) != null && Objects.equals(o1.get(field), o2.get(field));
+  static boolean equalsAndNotNull(Object o1, Object o2) {
+    return o1 != null && Objects.equals(o1, o2);
   }
 
-  static boolean presentAndNotEquals(OccurrenceFeatures o1, OccurrenceFeatures o2, String field) {
-    return o1.get(field) != null && !Objects.equals(o1.get(field), o2.get(field));
+  static boolean presentAndNotEquals(Object o1, Object o2) {
+    return o1 != null && o2 != null && !Objects.equals(o1, o2);
   }
 
-  static boolean presentOnOneOnly(OccurrenceFeatures o1, OccurrenceFeatures o2, String field) {
-    return (o1.get(field) == null && o1.get(field) != null) || (o1.get(field) != null && o1.get(field) == null);
```

Comment:
This was seemingly wrong before the rector (all `o1` and no `o2`) and fixed here.